test_util/test_upgrade_vpc.py

```diff
@@ -473,6 +590,21 @@ def apply(self, dcos_url: str, masters: List[str], public_masters: List[str], sl
                               default_os_user, DcosUser(CI_CREDENTIALS))
 
 
+class VpcClusterUpgradeTest(VpcClusterUpgradeTestFactory):
+    def mesos_metrics_snapshot(self, cluster, host):
```

Comment:
This should stay a part of the `Cluster` class. We'll likely find other reasons to get a Mesos metrics snapshot aside from the upgrade test. Also the knowledge of which port Mesos masters listen on vs. agents and the URL for the metrics endpoint fit better in the `Cluster` class.